src/components/taskDetails/index.tsx

```diff
@@ -62,7 +62,12 @@ type Props = {
 
 const TaskDetails: FC<Props> = ({ url, taskID }) => {
     const router = useRouter();
-    const isAuthorized = useContext(isUserAuthorizedContext);
+    // const [isUserAuthorized, setIsUserAuthorized] = useState<boolean>(false);
+    // const isAuthorized = useContext(isUserAuthorizedContext);
+    const userData = useSelector((state: any) => state.user);
+    const adminData = userData.adminUser;
+    const superUserData = userData.superUser;
+    const isAuthorized = !!adminData || !!superUserData;
```

Comment:
now we are repeating this logic on every page, we need to find a better way to do this

Reply:
On it! . Can I get an extension for the due date?